app/controllers/middleware_server_controller.rb

```diff
@@ -273,22 +292,26 @@ def run_server_operation(operation_info, items)
   end
 
   def trigger_mw_operation(operation, mw_server, params = nil)
-    mw_manager = mw_server.ext_management_system
-    path = mw_server.ems_ref
+    if operation == :generate_jdr
```

Comment:
Callers are `run_server_operation` and `run_server_params_operation`. There are some conditionals on the callers. So, I think you are right. I will move the code to callers.